### ISSUE TITLE
API Token Without IP Restriction

### ISSUE THREAD

**Issue by Aarondorn2** [eveseat/seat#186]
* **Problem**: The SeAT API is not accessible from applications that have dynamic IPs.  For instance, I'm attempting to post an API key/ID on a Google Form Submit to SeAT.  The Google script will use an IP from the following ranges:

216.239.32.0 - 216.239.63.255
64.233.160.0 - 64.233.191.255
66.249.80.0 - 66.249.95.255
72.14.192.0 - 72.14.255.255
209.85.128.0 - 209.85.255.255
66.102.0.0 - 66.102.15.255
74.125.0.0 - 74.125.255.255
64.18.0.0 - 64.18.15.255
207.126.144.0 - 207.126.159.255
173.194.0.0 - 173.194.255.255

There is no way to allow access to the API this way.

* **Expected**: One solution would be to allow the user to create a token without an IP restriction (less secure).  Another might be to allow multiple IPs and/or ranges for a single token.

**Comment by warlof** [comment]
Hello Aarondor2,

We already get the report early in january and think about implementing CIDR notation and multiple IP whitelisting for a single token.

In order to achieve this, we've decided to use a JSON formatted value stored into the current api_tokens table. This choice will allow us to reduce the impact of such change.
In addition, we've think to use this library https://packagist.org/packages/s1lentium/iptools which appears us as a good candidate in order to handle IP check.

It's providing just the feature we need : IP range check, CIDR notation and IP validation. It's also supporting IPv4 and IPv6.

I've started change on this branch if you want to have a look https://github.com/warlof/api/tree/api-cidr

Unfortunately, I'm a bit busy at the moment with other third party work as well as IRL stuff.

As temporary solution, you're able to comment this line into `seatpath/vendor/eveseat/api/src/Http/Middleware/ApiToken.php`

https://github.com/eveseat/api/blob/master/src/Http/Middleware/ApiToken.php#L70

This file is handling IP check foreach API request.
Remember that everytime you will apply a patch for this package, you've to reapply the change until it's been released.
